chain/network/src/tests/data.rs

```diff
@@ -0,0 +1,288 @@
+        prev.header().height() + 5,                            // height
+        prev.header().block_ordinal() + 1,                     // block_ordinal
+        chunks.into_iter().map(|c| c.take_header()).collect(), // chunks
+        EpochId::default(),                                    // epoch_id
+        EpochId::default(),                                    // next_epoch_id
+        None,                                                  // epoch_sync_data_hash
+        vec![],                                                // approvals
+        Rational::from_integer(0),                             // gas_price_adjustment_rate
+        0,                                                     // min_gas_price
+        0,                                                     // max_gas_price
+        Some(0),                                               // minted_amount
+        vec![],                                                // challenges_result
+        vec![],                                                // challenges
+        signer,
+        CryptoHash::default(), // next_bp_hash
+        CryptoHash::default(), // block_merkle_root
+        Some(clock.utc_now()), // timestamp_override
+    )
+}
+
+pub fn make_account_id<R: Rng>(rng: &mut R) -> AccountId {
+    format!("account{}", rng.gen::<u32>()).parse().unwrap()
+}
+
+pub fn make_signer<R: Rng>(rng: &mut R) -> InMemorySigner {
+    let account_id = make_account_id(rng);
+    InMemorySigner::from_seed(account_id.clone(), KeyType::ED25519, &account_id)
+}
+
+pub fn make_validator_signer<R: Rng>(rng: &mut R) -> InMemoryValidatorSigner {
+    let account_id = make_account_id(rng);
+    InMemoryValidatorSigner::from_seed(account_id.clone(), KeyType::ED25519, &account_id)
+}
+
+pub fn make_addr<R: Rng>(rng: &mut R) -> SocketAddr {
+    SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::LOCALHOST, rng.gen()))
+}
+
+pub fn make_peer_info<R: Rng>(rng: &mut R) -> PeerInfo {
+    let signer = make_signer(rng);
+    PeerInfo {
+        id: PeerId::new(signer.public_key),
+        addr: Some(make_addr(rng)),
+        account_id: Some(signer.account_id),
+    }
+}
+
+pub fn make_announce_account<R: Rng>(rng: &mut R) -> AnnounceAccount {
+    let peer_id = PeerId::new(make_signer(rng).public_key);
+    let validator_signer = make_validator_signer(rng);
+    let signature = validator_signer.sign_account_announce(
+        validator_signer.validator_id(),
+        &peer_id,
+        &EpochId::default(),
+    );
+    AnnounceAccount {
+        account_id: validator_signer.validator_id().clone(),
+        peer_id: peer_id,
+        epoch_id: EpochId::default(),
+        signature,
+    }
+}
+
+pub fn make_partial_edge<R: Rng>(rng: &mut R) -> PartialEdgeInfo {
+    let a = make_signer(rng);
+    let b = make_signer(rng);
+    PartialEdgeInfo::new(
+        &PeerId::new(a.public_key),
+        &PeerId::new(b.public_key),
+        rng.gen(),
+        &a.secret_key,
+    )
+}
```

Comment:
Consider using https://docs.rs/arbitrary/latest/arbitrary/ for generating random data. 

If we want to stop at a single randomized test here, using rand is fine. If we want to grow this infrae to a long term randomizied testing infrastructure, I would feel rather strongly that we should base the thing on arbitrary. 

arbitrary is like rand, but:

* it is a _finite_ PRNG, so it's easy to vary length of input to vary the length of the test
* it can be easily plugged into coverage-guided fuzzing
* it provides derive for simple cases
* it's a better crate than rand overall (simpler API, no generics(compile time), no dependencies)

Reply:
What if I need more parameters than just an entropy source to construct test input? Like consistent clock timestamps for example?

Reply:
> a different distribution of input data might be desirable.

Indeed, and that's feasible with arbitrary: you don't *have* to implement Arbitrary trait, the main value comes from the `Unstructured` type:

```rust
fn make_strings(u: &mut arbitrary::Unstructured<'_>, alphabet: &str) -> arbitrary::Result<Vec<String>>
```

Reply:
https://docs.rs/wasm-smith/latest/wasm_smith/ would be a cool example of using highly-skewed distributions, esp SwarmConfig: https://docs.rs/wasm-smith/latest/wasm_smith/struct.SwarmConfig.html